src/commands/basemaps-github/create-pr.ts

```diff
@@ -61,6 +61,32 @@ async function parseTargetInfo(
   return { name: standardizeLayerName(name), epsg: epsg.code, title, region };
 }
 
+/**
+ * Pass vector target location with the following format
+ * s3://linz-basemaps-staging/vector/3857/53382-nz-roads-addressing/01HSF04SG9M1P3V667A4NZ1MN8/53382-nz-roads-addressing.tar.co
+ * s3://linz-basemaps-staging/vector/3857/topographic/01HSF04SG9M1P3V667A4NZ1MN8/topographic.tar.co
+ */
+async function parseVectorTargetInfo(target: string): Promise<{ name: string; title: string; epsg: EpsgCode }> {
+  logger.info({ target }, 'CreatePR: Get the layer information from target');
+  const url = new URL(target);
+  const bucket = url.hostname;
+  const splits = url.pathname.split('/');
+  const epsg = Epsg.tryGet(Number(splits[2]));
+  const name = splits[3];
```

Comment:
A lot of this functionality looks like it's a copy of `parseRasterTargetInfo`. Would it be possible to extract a method to reuse most of it?

Reply:
I extracted the duplicated logic into `assertValidBucket` function.